autotest/cpp/test_gdal_pixelfn.cpp

```diff
@@ -0,0 +1,262 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// Project:  C++ Test Suite for GDAL/OGR
+// Purpose:  Test constant and builtin arguments for C++ pixel functions
+// Author:   Momtchil Momtchev <momtchil@momtchev.com>
+//
+///////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2022, Momtchil Momtchev <momtchil@momtchev.com>
+/*
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "gdal_unit_test.h"
+
+#include "cpl_string.h"
+#include "gdal_alg.h"
+#include "gdal_priv.h"
+#include "gdal.h"
+#include "../frmts/vrt/vrtdataset.h"
```

Comment:
like without effect as this would resolve to autotest/frmts/vrt/vrtdataset.h

Reply:
It does build without it, isn't this relative to the directory you are building from?

Reply:
> isn't this relative to the directory you are building from?

I would have said relative to file, but in any case the build directory is autotest/cpp, so that's equivalent. Actually, this only works because there are include paths like -I/home/even/gdal/gdal/gcore which make things eventually work.

Reply:
I couldn't make it work with `vrt/vrtdataset.h` - what do you suggest?

Reply:
let's keep it that way ( #include "../../frmts/vrt/vrtdataset.h" would also work )